validator/utils/synthetic/synthetic_utils.py

```diff
@@ -109,22 +97,37 @@ async def _get_random_picsum_image(x_dim: int, y_dim: int) -> str:
     return img_b64
 
 
-async def get_random_image_b64(cache: diskcache.Cache) -> str:
-    for key in cache.iterkeys():
-        image_b64: str | None = cache.get(key, None)  # type: ignore
-        if image_b64 is None:
-            cache.delete(key)
-            continue
 
-        if random.random() < 0.01:
-            cache.delete(key)
-        return image_b64
 
+async def get_random_image_b64(redis_db: Redis) -> str:
+    cache_key = scst.IMAGE_CACHE_KEY
+    cache_size = scst.IMAGE_CACHE_SIZE
+
+    logger.debug(f"Starting get_random_image_b64. Cache key: {cache_key}, Cache size: {cache_size}")
+
+    if random.random() < 0.01:
+        if await redis_db.scard(cache_key) > 0:
```

Comment:
This could maybe be a function as it's not clear to me what this does without reading line by line and trying to work it out.


if random.random() < 0.01:
 _do_some_logic()
etc